### DISCUSSION TITLE
/command/set_charging_amps?charging_amps={value}  fails with missing charging_amps param

### DISCUSSION THREAD

**Discussion by jgelsey** [timdorr/tesla-api#723] [Q&A] [answered]
Any help greatly appreciated.  Here's an example - doesn't matter what `{value}` is set to, I've tried multiple integers from 5 to 32.

`curl --location -g --request POST 'https://owner-api.teslamotors.com/api/1/vehicles/{id}/command/set_charging_amps?charging_amps={**value**}' \
--header 'Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6Im5ZdVZJWTJTN3gxVHRYM01KMC1QMDJad3pXQSJ9.eyJpc3MiOiJodHRwczovL2F1dGgudGVzbGEuY29tL29hdXRoMi92MyIsImF1ZCI6WyJodHRwczovL293bmVyLWFwaS50ZXNsYW1vdG9ycy5jb20vIiwiaHR0cHM6Ly9hdXRoLnRlc2xhLmNvbS9vYXV0aDIvdjMvdXNlcmluZm8iXSwiYXpwIjoib3duZXJhcGkiLCJzdWIiOiI1OWJjOGVhNC01ZjdjLTQwZDUtODE0YS0xZWNkOTc2NTIxZTkiLCJzY3AiOltdLCJhbXIiOltdLCJleHAiOjE2ODUzNjI5MDksImlhdCI6MTY4NTMzNDEwOSwib3VfY29kZSI6Ik5BIn0.TBlS58uBqs1S9VbRu6IXmDnGJnPKfmR8-Au8T5So8r1Dz7lOSym-ldx_BRWlXfOGXT2des3pqWY_6jQR_TYLc6oceRuJ34bOI0vzintu26QgsqbWz77oEjxPuBNBlZ3BBHIq9sQp9FU8aigFUG7SsKmAxzf8Gz5XegygotarQQctiTJWcg_9mOsW8BJk8N9H5iagRbQDStqbNMRJvFftGzN0dPI1ytGE4W2z4C6uK87TMTK06RSwwahSvhhgBhb4Z1d-_PJ_0yJsVm5L1hsA7kAN4AN-6XrojkdJQqfZPO8o9-fZz-Nwy0OIxyeasUXoGBvTE3k_HmcdMaSrZK2nOg'`

Error response is:

`{"response":{"result":false,"reason":"missing charging_amps param"}}% `

Bearer token and vehicle id are both correct, as they work with other calls.  

This is driving me crazy, since near as I can tell I'm following the documentation exactly.  Almost feels like there's a bug in Tesla's API that has a corrupted string it's comparing the `charging_amps` parameter to.

**Comment by GaPhi** [comment]
Parameter name is 'amps' not 'charging_amps'

**Reply by jgelsey** [reply]
Unfortunately using `amps` instead of `charging_amps` as the parameter makes no difference, the same error is returned: `{"response":{"result":false,"reason":"missing charging_amps param"}`.  Would you happen to have a working example?

**Reply by superfloh247** [reply]
you have to POST it, not GET it

**Reply by jgelsey** [reply]
Yes, that's what I show in the example in the original question, a `POST`.  Would you happen to have a working example?

**Comment by superfloh247** [comment, marked as answer]
URL command/set_charging_amps
payload {"charging_amps":12}

**Reply by jgelsey** [reply]
It works!  I was putting `"charging_amps"` as a query string rather than `--data` in my `curl`.   Many, many thanks!

**Comment by joeyh** [comment]
Huh, this doesn't work for me, I get back "Tesla Vehicle Command Protocol required, please refer to the documentation here: https://developer.tesla.com/docs/fleet-api#2023-10-09-rest-api-vehicle-commands-endpoint-deprecation-warning"

@superfloh247 is this still working for you?
